src/dwc.cpp

```diff
@@ -0,0 +1,533 @@
+#include "peripherals/dwc.h"
+
+#include "libk.h"
+#include "peripherals/timer.h"
+#include "printf.h"
+
+void usb_reset_controller() {
+    *USB_GRSTCTL = (1 << 0);
+
+    while (*USB_GRSTCTL & (1 << 0)) {
+    }
+
+    *USB_GUSBCFG = USB_FORCE_HOST_MODE;
+    *USB_HCFG |= (1 << 23);
+}
+
+void usleep(unsigned int microseconds) {
```

Comment:
fine for now but at some point we should move this to a timer class, i think we also have one in the SD driver pr im about to merge